bindings/nodejs/lib/types/block/output/token-scheme.ts

```diff
@@ -26,39 +26,59 @@ abstract class TokenScheme {
  * Simple token scheme.
  */
 class SimpleTokenScheme extends TokenScheme {
-    private mintedTokens: HexEncodedAmount;
-    private meltedTokens: HexEncodedAmount;
-    private maximumSupply: HexEncodedAmount;
+    private mintedTokens: bigint;
+    private meltedTokens: bigint;
+    private maximumSupply: bigint;
 
     constructor(
-        mintedTokens: HexEncodedAmount,
-        meltedTokens: HexEncodedAmount,
-        maximumSupply: HexEncodedAmount,
+        mintedTokens: bigint,
+        meltedTokens: bigint,
+        maximumSupply: bigint,
     ) {
         super(TokenSchemeType.Simple);
-        this.mintedTokens = mintedTokens;
-        this.meltedTokens = meltedTokens;
-        this.maximumSupply = maximumSupply;
+        if (typeof mintedTokens === 'bigint') {
+            this.mintedTokens = mintedTokens;
+        } else if (mintedTokens) {
+            this.mintedTokens = hexToBigInt(mintedTokens);
+        } else {
+            this.mintedTokens = BigInt(0);
+        }
+
+        if (typeof meltedTokens === 'bigint') {
```

Comment:
SHould we make this a helper function? we have it a few times around

Reply:
what exactly? the `typeof meltedTokens === 'bigint'` part?

Reply:
the `if {} elseif {} else {}`

Reply:
this one is not really duplicated anywhere else, but there's 
```typescript 
        if (params.amount && typeof params.amount === 'bigint') {
            params.amount = params.amount.toString(10);
        }
```
duplicated in a couple of places, but maybe it's better to keep it as is, since it's not much and and we would avoid modifying the object by reference in a function.

Reply:
i mena like a mothod:
```
funciton coerceBigInt(item) -> BigInt
       if (typeof item=== 'bigint') {
           return item;
        } else if (item) {
            return hexToBigInt(item);
        } else {
            return BigInt(0);
        }
```